packages/openmrs/src/Adaptor.js

```diff
@@ -162,101 +150,86 @@ export function get(path, query, callback = false) {
  * @param {function} [callback] - Optional callback to handle the response
  * @returns {Operation}
  */
-export function post(path, data, callback = false) {
-  return state => {
-    const resolvedPath = expandReferences(path)(state);
-    const resolvedData = expandReferences(data)(state);
-    const { instanceUrl } = state.configuration;
-
-    const urlPath = `${instanceUrl}/ws/rest/v1/${resolvedPath}`;
-
-    return agent
-      .post(urlPath)
-      .type('json')
-      .send(resolvedData)
-      .then(response => handleResponse(response, state, callback))
-      .catch(handleError);
+export function post(path, data, callback = s => s) {
+  return async state => {
+    const [resolvedPath, resolvedData] = expandReferences(state, path, data);
+
+    console.log(`Sending an HTTP post to ${resolvedPath}`);
```

Comment:
@josephjclark  Should I display the whole URL?

Reply:
Ooh good question. Probably just the path but maybe not 🤔

Pause on this bit for now, because it's affected by my other comment  (the one I'll get back to you for)